apps/learning-snippets/src/test/tree-widget/Telemetry.test.tsx

```diff
@@ -0,0 +1,233 @@
+              return { category };
+            })
+          ).imodel;
+          const testViewport = getTestViewer(testImodel);
+          const unifiedSelectionStorage = createStorage();
+          sinon.stub(IModelApp.viewManager, "selectedView").get(() => testViewport);
+          sinon.stub(UiFramework, "getIModelConnection").returns(testImodel);
+          const consoleSpy = sinon.spy(console, "log");
+
+          // __PUBLISH_EXTRACT_START__ TreeWidget.TelemetryTreeComponentExample
+          function MyWidget() {
+            return (
+              <IModelContentTreeComponent
+                onPerformanceMeasured={(feature, elapsedTime) => {
+                  console.log(`TreeWidget [${feature}] took ${elapsedTime} ms`);
+                }}
+                onFeatureUsed={(feature) => {
+                  console.log(`TreeWidget [${feature}] used`);
+                }}
+                getSchemaContext={getSchemaContext}
+                selectionStorage={unifiedSelectionStorage}
+              />
+            );
+          }
+          // __PUBLISH_EXTRACT_END__
+          render(<MyWidget />);
+
+          await waitFor(() => {
+            expect(consoleSpy).to.be.calledOnce;
+          });
+          cleanup();
+        });
+
+        it("Renders custom categories tree with telemetry", async function () {
+          const imodel = (
+            await buildIModel(this, async (builder) => {
+              const physicalModel = insertPhysicalModelWithPartition({ builder, codeValue: "TestPhysicalModel" });
+              const category = insertSpatialCategory({ builder, codeValue: "Test SpatialCategory" });
+              insertPhysicalElement({ builder, modelId: physicalModel.id, categoryId: category.id });
+              return { category };
+            })
+          ).imodel;
+          const viewport = getTestViewer(imodel);
+          const unifiedSelectionStorage = createStorage();
+          sinon.stub(IModelApp.viewManager, "selectedView").get(() => imodel);
+          sinon.stub(UiFramework, "getIModelConnection").returns(imodel);
+          const consoleSpy = sinon.spy(console, "log");
+
+          // __PUBLISH_EXTRACT_START__ TreeWidget.TelemetryCustomTreeExample
+          function MyWidget() {
+            return (
+              <TelemetryContextProvider
+                componentIdentifier="MyTree"
+                onPerformanceMeasured={(feature, elapsedTime) => {
+                  console.log(`TreeWidget [${feature}] took ${elapsedTime} ms`);
+                }}
+                onFeatureUsed={(feature) => {
+                  console.log(`TreeWidget [${feature}] used`);
+                }}
+              >
+                <MyTree />
+              </TelemetryContextProvider>
+            );
+          }
+          function MyTree() {
+            const { categoriesTreeProps, rendererProps } = useCategoriesTree({ activeView: viewport, filter: "" });
+
+            return (
```

Comment:
I think this will improve readability:

```suggestion
          }

          function MyTree() {
            const { categoriesTreeProps, rendererProps } = useCategoriesTree({ activeView: viewport, filter: "" });
            return (
```